Add support for DNS SRV lookups

mhamann opened the pull request.

mhamann commented:
@michaelshobbs it took me far longer to complete this than I originally intended, but at last, it is ready to go. :-)

michaelshobbs commented:
No worries. Thanks for the contribution

mhamann commented:
@michaelshobbs what do you need before this can be merged?

ChrisLahaye commented:
@michaelshobbs can this get merged please?
@mhamann are you still using service discovery in logspout, what are you using instead?

michaelshobbs commented:
this PR has merge conflicts. happy to merge once the feedback and conflicts are resolved.

mhamann commented:
@ChrisLahaye I've since moved my environment to Kubernetes. Between k8s services and kube-dns, I no longer need SRV lookups.

I believe I'd originally resolved conflicts and addressed review comments, but I guess it must've fallen through the cracks. Not sure I'll have time to fix this up again, but would be happy for someone else to handle it. Seems like it would still be a good contribution.
